pkg/v3/types/basetypes.go

```diff
@@ -163,6 +163,21 @@ type CheckResult struct {
 	RetryInterval time.Duration
 }
 
+// Size returns the size of the check result in bytes
+func (cr *CheckResult) Size() int {
```

Comment:
Note that all the numbers have a corresponding comment to specify what field they represent.
The most naive option is to encode the object and count the bytes, but that would be expensive in runtime (encoding/decoding is currently taking dozens percentage of CPU)

The CheckResult currently still resides here, once moved to cl-common I'll adjust.

Reply:
+1 don't quite like this hardcoded size counting as historically this as bitten us in the past. 

1. Anytime the struct is changed it creates an implicit dependency to change which is very easy to forget
2. If encoding type is changed then this needs to change
3. This will likely need some magic numbers to match exact json encoding size then just adding the byte size of fields